tractor/_trionics.py

```diff
@@ -53,9 +57,34 @@ def __init__(self, actor, supervisor=None):
         )
         parent_addr = self._actor.accept_addr
         assert parent_addr
+        self._forkserver = fs = forkserver._forkserver
+        if mp.current_process().name == 'MainProcess' and (
+            not self._actor._forkserver_info
+        ):
+            # if we're the "main" process start the forkserver only once
+            # and pass it's ipc info to downstream children
+
+            # forkserver.set_forkserver_preload(rpc_module_paths)
+            forkserver.ensure_running()
+            fs_info = addr, alive_fd, pid, st_pid, st_fd = (
+                fs._forkserver_address,
+                fs._forkserver_alive_fd,
+                getattr(fs, '_forkserver_pid', None),
+                getattr(semaphore_tracker._semaphore_tracker, '_pid', None),
+                semaphore_tracker._semaphore_tracker._fd,
+            )
+        else:
+            fs_info = (
+                fs._forkserver_address,
+                fs._forkserver_alive_fd,
+                fs._forkserver_pid,
+                semaphore_tracker._semaphore_tracker._pid,
+                semaphore_tracker._semaphore_tracker._fd,
+             ) = self._actor._forkserver_info
```

Comment:
These two assignments are kinda tricky: you're building a tuple, unpacking tuples two different ways, while also chaining assignment. Its clever, but maybe borderline not readable?

At the very least, I could see someone assuming these two statements as similar - as they're written to look parallel - but they're not. The dominating block of the top constructs a tuple, while its unpacking one in the bottom. That last assignment could be visually easy to overlook.

Personal rule of thumb (imho) - the further you get to the end of a line, the less important the details should be.

I'd consider rewriting this, even if it makes it more verbose - but I don't necessarily consider this an egregiously offense either.